src/sentry/search/events/datasets/discover.py

```diff
@@ -677,6 +677,109 @@ def function_converter(self) -> Mapping[str, SnQLFunction]:
+                                                                                                        args[
+                                                                                                            "spans_group"
+                                                                                                        ],
+                                                                                                    ],
+                                                                                                ),
+                                                                                            ],
+                                                                                        ),
+                                                                                    ),
+                                                                                    Column(
+                                                                                        "spans.exclusive_time"
+                                                                                    ),
+                                                                                    Column(
+                                                                                        "spans.op"
+                                                                                    ),
+                                                                                    Column(
+                                                                                        "spans.group"
+                                                                                    ),
+                                                                                ],
+                                                                            )
+                                                                        ],
+                                                                    ),
+                                                                    args["multiplier"],
+                                                                ],
+                                                            ),
+                                                            args["start_offset"],
+                                                        ],
+                                                    ),
+                                                    args["bucket_size"],
+                                                ],
+                                            ),
+                                        ],
+                                    ),
+                                    args["bucket_size"],
+                                ],
+                            ),
+                            args["start_offset"],
+                        ],
+                        alias,
+                    ),
```

Comment:
As I understand it the first set of functions match the existing histogram function, ie. both are 
`floor((x * multiplier - start_offset) / bucket_size) * bucket_size + start_offset`
but the `x` for spans_histogram needs to be handled differently than plain `histogram`

I wonder if we could introduce a helper function (eg. something like `_resolve_histogram`) that would take a param that's the x to reduce code repeat, which would mean that this function could be updated to something like:
```
                    snql_column=self._resolve_histogram(
                        Function(
                            "arrayJoin",
                            [
                                Function(
                                    "arrayFilter",
                                    [
                                        Lambda(
                                            [
                                                "z",
                                                "x",
                                                "y",
                                            ],
                                            Function(
                                                "and",
                                                [
                                                    Function(
                                                        "equals",
                                                        [
                                                            Identifier(
                                                                "x"
                                                            ),
                                                            args[
                                                                "spans_op"
                                                            ],
                                                        ],
                                                    ),
                                                    Function(
                                                        "equals",
                                                        [
                                                            Identifier(
                                                                "y",
                                                            ),
                                                            args[
                                                                "spans_group"
                                                            ],
                                                        ],
                                                    ),
                                                ],
                                            ),
                                        ),
                                        Column(
                                            "spans.exclusive_time"
                                        ),
                                        Column(
                                            "spans.op"
                                        ),
                                        Column(
                                            "spans.group"
                                        ),
                                    ],
                                )
                            ],
                        ),
```

This also means we can move the comment explaining the math to the helper function too since it applies in both scenarios

Reply:
I'd probably only DRY on the 3rd time....

Reply:
I'll revisit this down the line 👍